### PR TITLE
Add DataFrames.jl extension

### PR DESCRIPTION
**Pull request by odow**
Take II of #3441
x-ref #3438 

TODOs

 - [x] Add documentation to extension
 - [x] Rework existing tutorials to leverage new syntax

See https://jump.dev/JuMP.jl/previews/PR3821/extensions/DataFrames/

### CODE REVIEW

File: docs/src/tutorials/linear/multi.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # v.2.0. If a copy of the MPL was not distributed with this file, You can       #src
 # obtain one at https://mozilla.org/MPL/2.0/.                                   #src
 
-# # The multi-commodity flow problem
+# # Working with SQLite and DataFrames
```
Comment (odow) [inline docs/src/tutorials/linear/multi.jl:6]:
```suggestion
# # The multi-commodity flow problem
```

**Comment by odow** [comment]
Is this an improvement? I'm not sure.

Now: https://jump.dev/JuMP.jl/previews/PR3821/tutorials/linear/multi/
Previous: https://jump.dev/JuMP.jl/stable/tutorials/linear/multi/

File: docs/src/extensions/DataFrames.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+     │ String  Constrai…
+─────┼──────────────────────────────────────
+   1 │ a       x[2,a] + x[3,a] + x[4,a] ≤ 1
+   2 │ b       x[2,b] + x[3,b] + x[4,b] ≤ 1
+```
+
+### DataFrame-native syntax
+
+While you can use indexing in JuMP's `@expression` and `@constraint` macros, it
+may be more convienent to use DataFrames.jl split-apply-combine framework. For
```
Comment (odow) [inline docs/src/extensions/DataFrames.md:117]:
```suggestion
may be more convenient to use DataFrames.jl split-apply-combine framework. For
```

**Comment by blegat** [comment]
We could also have an array with the same syntax as `SparseAxisArray` and a `DataFrame` instead of a `Dict` as a backend.
I guess we could also have both the container of your PR and the one I just suggested.

File: docs/src/extensions/DataFrames.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+and the constraint could be written as
+
+```jldoctest ext_data_frames
+julia> df_constraint(v) = @constraint(model, sum(v) <= 1);
+
+julia> DataFrames.combine(
+           DataFrames.groupby(x, :j),
+           :value => df_constraint => :value,
+       )
```
Comment (joaquimg) [inline docs/src/extensions/DataFrames.md:141]: This call the in-place operations?

Comment (odow) [reply docs/src/extensions/DataFrames.md:141]: I don't understand the question. It's not in-place. It creates a new DataFrame

**Pull request closed without merging**